pkgs/build-support/emacs/generic.nix

```diff
@@ -60,8 +60,35 @@ stdenv.mkDerivation ({
 
   LIBRARY_PATH = "${lib.getLib stdenv.cc.libc}/lib";
 
+  addEmacsNativeCompPath = true;
+
+  dontPatchELF = true;
+  dontStrip = true;
+
   postInstall = ''
-    find $out/share/emacs -type f -name '*.el' -print0 | xargs -0 -n 1 -I {} -P $NIX_BUILD_CORES sh -c "emacs --batch -f batch-native-compile {} || true"
+    find $out/share/emacs -type f -name '*.el' -print0 \
+      | xargs -0 -n 1 -I {} -P $NIX_BUILD_CORES sh -c \
+          "emacs --batch --eval=\"(add-to-list 'comp-eln-load-path \\\"$out/share/emacs/native-lisp/\\\")\" -f batch-native-compile {} || true"
+  '';
+
+  # Like patchELF in pkgs/development/tools/misc/patchelf/setup-hook.sh, except
+  # we skip native-comp binaries.
```

Comment:
I'm not sure this is necessary, but the `patchELF` hook is running on `.eln` files and I was concerned that would affect the file hash. Looking at this now, I think we can get rid of it as I believe the content hash is generated from the elisp file and not the resulting .eln.